GDJS/Runtime/pixi-renderers/pixi-object-effects-manager.ts

```diff
@@ -0,0 +1,169 @@
+/*
+ * GDevelop JS Platform
+ * Copyright 2013-2021 Florian Rival (Florian.Rival@gmail.com). All rights reserved.
+ * This project is released under the MIT License.
+ */
+namespace gdjs {
+  import PIXI = GlobalPIXIModule.PIXI;
+
+  class PixiObjectEffectsManager {
+
+    update(runtimeObject: RuntimeObject, layer: Layer) {
+      const filters = runtimeObject.getFilters();
+      for (const filterName in filters) {
+        const filter = filters[filterName];
+        filter.update(filter.pixiFilter, layer);
+      }
+    }
+
+    /**
+     * Add a new effect on a runtime object, or replace the one
+     * with the same name.
+     * @param runtimeObject The runtime object
+     * @param effectData The data of the effect to add.
+     */
+    addEffect(
+      runtimeObject: RuntimeObject,
+      effectData: EffectData,
+      layer: Layer
+    ) {
+      const filterCreator = gdjs.PixiFiltersTools.getFilterCreator(
+        effectData.effectType
+      );
+      if (!filterCreator) {
+        console.log(
+          'Filter "' +
+            effectData.name +
+            '" has an unknown effect type: "' +
+            effectData.effectType +
+            '". Was it registered properly? Is the effect type correct?'
+        );
+        return;
+      }
+
+      const filter: PixiFiltersTools.Filter = {
+        pixiFilter: filterCreator.makePIXIFilter(layer, effectData),
+        updateDoubleParameter: filterCreator.updateDoubleParameter,
+        updateStringParameter: filterCreator.updateStringParameter,
+        updateBooleanParameter: filterCreator.updateBooleanParameter,
+        update: filterCreator.update,
+      };
+
+      const renderer = runtimeObject.getRendererObject();
+      renderer.filters = (renderer.filters || []).concat(filter.pixiFilter);
+
+      const filters = runtimeObject.getFilters();
+      filters[effectData.name] = filter;
+    }
+
+    /**
+     * Remove the effect from a runtime object with the specified name
+     * @param runtimeObject The runtime object.
+     * @param effectName The name of the effect.
+     */
+    removeEffect(runtimeObject: RuntimeObject, effectName: string) {
+      const filter = runtimeObject.getFilters()[effectName];
+      if (!filter) return;
+      const renderer: PIXI.DisplayObject = runtimeObject.getRendererObject();
+      renderer.filters = (renderer.filters || []).filter(
+        (pixiFilter) => pixiFilter !== filter.pixiFilter
+      );
+      delete filter[effectName];
```

Comment:
You're sure about this line? `filter` is already a filter object, it does not have any `effectName` property.
You probably want to run delete on `runtimeObject.getFilters()`

Reply:
Oh yes, right. I think at this point I should be adding some tests to avoid these silly mistakes. 😬

Reply:
Might be a good idea to think about unit/integration tests (probably integration tests if this requires PixiJS) yes, to check the good working of methods that add/change/delete effects :)